dotCMS/src/main/java/com/dotmarketing/portlets/workflows/business/WorkflowAPIImpl.java

```diff
@@ -1784,17 +1847,35 @@ public WorkflowAction findActionRespectingPermissions(final String actionId,
         return action;
     }
 
+	/**
+	 * Returns the actions associted to the content type if there's no license. only actions
+	 * associated with the system workflow will be returned.
+	 *
+	 * @param contentType ContentType to be processed
+	 * @param user The current User
+	 */
 	@CloseDBIfOpened
-	public List<WorkflowAction> findAvailableDefaultActionsByContentType(ContentType contentType, User user)
+	public List<WorkflowAction> findAvailableDefaultActionsByContentType(ContentType contentType,
+			User user)
 			throws DotDataException, DotSecurityException {
 
 		DotPreconditions.isTrue(this.permissionAPI.
-						doesUserHavePermission(contentType, PermissionAPI.PERMISSION_READ, user,true),
+						doesUserHavePermission(contentType, PermissionAPI.PERMISSION_READ, user, true),
 				() -> "User " + user + " cannot read content type " + contentType.name(),
 				DotSecurityException.class);
 
 		List<WorkflowScheme> schemes = findSchemesForContentType(contentType);
-		return findAvailableDefaultActionsBySchemes(schemes, APILocator.getUserAPI().getSystemUser());
+
+		if (!hasValidLicense()) {
+			//When no License, we only allow the system workflow
+			final WorkflowScheme systemWorkflow = schemes.stream()
+					.filter(WorkflowScheme::isSystemWorkflowScheme).findFirst()
+					.orElse(workFlowFactory.findSystemWorkflow());
+			schemes = new ImmutableList.Builder<WorkflowScheme>().add(systemWorkflow).build();

```

Comment:
I think it could be a bit more efficient if you do
```


		if (!hasValidLicense()) {
			//When no License, we only allow the system workflow
			
			return new ImmutableList.Builder<WorkflowScheme>().add(workFlowFactory.findSystemWorkflow()).build();

		}

final List<WorkflowScheme> schemes = findSchemesForContentType(contentType);
....
```

Reply:
Given that we already have schemes I think it's worth to shot trying to find the system workflow before attempting another hit to the db